api/middleware/cache.js

```diff
@@ -0,0 +1,18 @@
+const middlewareCache = {};
+const cache = (key, getMiddleware) => {
+  if (!middlewareCache[key]) {
+    middlewareCache[key] = getMiddleware();
+  }
+  return middlewareCache[key];
+};
+
+const knownModels = [];
+const modelIndex = model => {
```

Comment:
take or leave this suggestion, but i would tweak this to have only one return statement:

```
const modelIndex = model => {
  if (!knownModels.includes(model)) knownModels.push(model);
  return knownModels.indexOf(model);
};
```

Reply:
I'm 100% in favor of this recommendation.